chains/ethereum/writer_methods.go

```diff
@@ -151,23 +162,27 @@ func (w *writer) createGenericDepositProposal(m msg.Message) bool {
 	return true
 }
 
+// watchThenExecute watches for the latest block and executes once the matching event is found
 func (w *writer) watchThenExecute(m msg.Message, handler common.Address, data []byte, latestBlock *big.Int) {
 	w.log.Trace("Watching for finalization event", "source", m.Source, "dest", m.Destination, "nonce", m.DepositNonce)
 
+	// waiting to connect to the block
```

Comment:
This feels out of place, "connect" isn't really the right verb and the loop itself isn't waiting. 

We have a loop to limit how many blocks we wait for before giving up. The connection is always maintained, the data just isn't available. The `waitForBlock` call ensures that the block is available before we try to fetch it by asking the node it's connected to what the current block is. Happy to expand on either part if you would like.

Reply:
what is lastestBlock and why do we look for it?

Reply:
We expect multiple relayers to vote on the proposal simultaneously, but there may be some time between when the first relayer creates the proposal and the last relayer votes to pass it. So, here we track the latest block on the chain, and search through several of them as they become available to look for the event the signal the proposal has passed. Block numbers are sequential, so we can just increment as we go.